x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/SinkOperator.java

```diff
@@ -23,6 +23,15 @@ public final Page getOutput() {
         throw new UnsupportedOperationException();
     }
 
+    protected abstract void doAddInput(Page page);
+
+    @Override
+    public final void addInput(Page page) {
+        // We need to change the ownership of the blocks of the input page before passing them to another driver.
+        page.allowPassingToDifferentDriver();
```

Comment:
++ yes, it's important to transfer ownership in the executing thread *before* the page leaves.